### ISSUE TITLE
Issues with multioutput regressions

### ISSUE THREAD

**Issue by romanovzky** [rasbt/mlxtend#387]
Hi there,

I'm trying to fit a stacked regressor with multioutputs (2 to be precise). I have the following error in training phase::

```python
~/NLP/venv/lib/python3.6/site-packages/mlxtend/regressor/stacking_cv_regression.py in fit(self, X, y, groups)
    172                 instance.fit(X[train_idx], y[train_idx])
    173                 y_pred = instance.predict(X[holdout_idx])
--> 174                 meta_features[holdout_idx, i] = y_pred
    175 
    176         # save meta-features for training data

ValueError: shape mismatch: value array of shape (27,2) could not be broadcast to indexing result of shape (27,)
```

which seems to be some sort of issue with having `n_estimators*n_outputs` shape of the meta-estimator features. Could you confirm this, or am I missing something obvious?

Cheers

**Comment by romanovzky** [comment]
Steps to reproduce:
```python
from sklearn.linear_model import LinearRegression, Lars
from mlxtend.regressor import StackingCVRegressor
from sklearn.datasets import make_regression
X, y = make_regression(n_samples=100, n_features=10, n_targets=2)
meta = LinearRegression()
base = [LinearRegression(), Lars()]
sr = StackingCVRegressor(regressors=base, meta_regressor=meta)
sr.fit(X,y)
```

Gets you

```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-14-f4b8211e4c6a> in <module>()
----> 1 sr_1.fit(X,y)

~/git/NLP/venv/lib/python3.6/site-packages/mlxtend/regressor/stacking_cv_regression.py in fit(self, X, y, groups)
    172                 instance.fit(X[train_idx], y[train_idx])
    173                 y_pred = instance.predict(X[holdout_idx])
--> 174                 meta_features[holdout_idx, i] = y_pred
    175 
    176         # save meta-features for training data

ValueError: shape mismatch: value array of shape (20,2) could not be broadcast to indexing result of shape (20,)
```

**Comment by rasbt** [comment]
Sorry about the confusion, I just see that the docstring of the StackingRegressor mentions target arrays of shape [n_samples] or [n_samples, n_targets], but it was currently only tested on arrays with `[n_samples]` shape.

> Could you confirm this, or am I missing something obvious?

You are not missing anything, it's likely not working because of the reason regarding the `n_estimators*n_outputs`. This should either be clarified in the docs, or (better) the support for [n_samples, n_targets] target arrays could be added

**Comment by romanovzky** [comment]
Cool, thanks!

In the meantime, I'm making my own class for my current problem, but I look forward to seeing this feature!

Cheers

**Comment by romanovzky** [comment]
Well, my class is quite complex as it does something very specific for a gargantuan pipeline I ended up making. Besides, this is for work, so I'm not allowed to pass on this specific piece of code (unless there's some company change about releasing code, which I'm not in the position to promise or change).

What I can do, given personal time, is to develop a bit of trimmed down, to-the-point, class. If I don't find the time, this is the general functionality I found to help around this issue:

CV_predictions method that for each estimator in the base layer get `shape=(n_observations, n_outputs)` predictions from the original data. I then concatenate the predictions from each regressor into a pandas dataframe, such that after all estimators were used to CV-predict the meta-features. I end up with a pandas dataframe of meta-features of the form `shape =(n_observations, n_estimators*n_outputs)` and then I further concatenate `n_output` columns for the `y_true`. Finally I used this pandas dataframe to get X_metafeatures, y_true to feed into the meta-regressor.

Long story short, collecting things in pandas dataframes helped me keeping track of dimensionality without having to think about it. This way I find my code to be easly changed to further outputs, or less, if needed.

I haven't had a proper look into your code, so I don't know how much of what I just said is silly, incompatible, or just stupid. I'll let you be the judge of that ;)
